coordinates.py

```diff
@@ -0,0 +1,117 @@
+"""
+This module contains Coordinates class, that inherit from tuple,
+add new methods to work with coordinate tuples
+"""
+from itertools import permutations as perm
+from collections import namedtuple
+from heapq import merge
+
+
+class Coordinates(tuple):
+    """
+    Extend standard tuple class implements new methods
+    to work with coordinate tuples
+    """
+    def __init__(self, hex_: tuple[int, int, int]):
+        if not isinstance(hex_, tuple):
+            raise TypeError("Should pass a tuple of coordinates into Coordinates")
+        if not len(hex_) == 3:
+            raise TypeError("Tuple should have exactly 3 coordinates")
+        if all(isinstance(i, int) for i in hex_):
+            super().__init__()
+        else:
+            raise TypeError("Coordinates should be integers")
+
+    def cube_distance(self, other) -> int:
+        """
+        :param other: Coordinates obj
+        :return: distance between hexes
+        """
+        return max(abs(i - j) for i, j in zip(self, other))
+
+    def offset(self, offset_: tuple[int, int, int]) -> "Coordinates":
+        """
+        :param offset_: coordinate offset tuple (dx, dy, dz)
+        :return: Coordinates obj
+        """
+        result = tuple(i + j for i, j in zip(self, offset_))
+        return Coordinates(result)
+
+    def neighbours(self) -> set["Coordinates"]:
+        """
+        :return: set of neighbours coordinate tuples
+        """
+        return set(self.offset(direction) for direction in perm(range(-1, 2), 3))
+
+    def in_radius(self, radius: int) -> set["Coordinates"]:
+        """
+        :param radius: radius of circle
+        :return: set of Coordinates obj in given radius from self, includes self
+        """
+        result = set()
+        for d_x in range(- radius, radius + 1):
+            for d_y in range(max(- radius, - d_x - radius), min(radius, - d_x + radius) + 1):
+                d_z = - d_x - d_y
+                result.add(self.offset((d_x, d_y, d_z)))
+        return result
+
+    def normal_directions(self, radius: int) -> list[set["Coordinates"], ...]:
+        """
+        :param radius: maximum cells on each direction
+        :return: list of sets of Coordinates for each normal direction for self, in given radius
+        """
+        x_positive_dir = {self.offset((delta, 0, -delta)) for delta in range(1, radius + 1)}
+        x_negative_dir = {self.offset((-delta, 0, delta)) for delta in range(1, radius + 1)}
+        y_positive_dir = {self.offset((-delta, delta, 0)) for delta in range(1, radius + 1)}
+        y_negative_dir = {self.offset((delta, -delta, 0)) for delta in range(1, radius + 1)}
+        z_positive_dir = {self.offset((0, -delta, delta)) for delta in range(1, radius + 1)}
+        z_negative_dir = {self.offset((0, delta, -delta)) for delta in range(1, radius + 1)}
+        return [x_positive_dir, x_negative_dir, y_positive_dir,
+                y_negative_dir, z_positive_dir, z_negative_dir]
```

Comment:
In my opinion such kind of code style is not really a good option because it's hard to extend and merge code formatted like this. I would rather split this list for multiple lines right after first square brace and ended it with trailing square brace. Like this:
```
return [
    x_positive_dir, x_negative_dir, y_positive_dir,
    y_negative_dir, z_positive_dir, z_negative_dir,
]
```
Or even like this (in my opinion this option is better in terms of extending):
```
return [
    x_positive_dir,
    x_negative_dir
    y_positive_dir,
    y_negative_dir,
    z_positive_dir,
    z_negative_dir,
]
```
But the best option is to use automatic formatter. For example black.